packages/firestore/src/api/timestamp.ts

```diff
@@ -95,6 +95,10 @@ export class Timestamp {
     );
   }
 
```

Comment:
Strictly speaking, this isn't necessary -- `Timestamp` would have the same JSON representation without this function. I mostly added it to make it explicit that `Timestamp` has a well-defined JSON representation -- let me know if you prefer to remove it.

By the way, why is it that `GeoPoint` uses different names for internal variables and getters while `Timestamp` doesn't?

Reply:
The implementation in GeoPoint is a little safer: the `readonly` attribute is Typescript-only, so nothing saves a JavaScript user from directly modifying `Timestamp.seconds` or `Timestamp.nanoseconds`. To do the same with a GeoPoint, they would have to use an underscore prefixed variable (which is likely mangled), which would obviously be something we don't support.

Why Timestamp doesn't use this pattern is a question I can't answer.